### ISSUE TITLE
No enum constant exception getting indexes for edge collection

### ISSUE THREAD

**Issue by dan-drew** [arangodb/arangodb#2301]
## Environment
I'm using the latest ArangoDB of the respective release series: **3.1**
On this operating system: **Linux, docker - official docker library, arangodb:3.1**
Using driver: **com.arangodb:arangodb-java-driver:4.1.7**

## Example code (Groovy)
Where "collection" is the edge collection (edge1 in the test log)
```
Collection<IndexEntity> indexes = collection.indexes
```

Collection info as JSON
```
{"type":"EDGES","status":"LOADED","isVolatile":null,"id":"81425","waitForSync":null,"name":"edge1","isSystem":false}
```

## Result
```
com.arangodb.ArangoDBException: com.arangodb.velocypack.exception.VPackParserException: java.lang.IllegalArgumentException: No enum constant com.arangodb.entity.IndexType.edge
```

## Test log
Note that DB and collection names have been replaced

```
Testing started at 12:35 PM ...
Connected to the target VM, address: '127.0.0.1:58580', transport: 'socket'
Connected to ArangoDB on 192.168.99.100
Open connection to addr=192.168.99.100,port=8529
Connected to Socket[addr=/192.168.99.100,port=8529,localport=58585]
Send velocystream protocol header to Socket[addr=/192.168.99.100,port=8529,localport=58585]
Send Message (id=1, head=[1,1,"_system",1,"\/_api\/database",{},{}], body={})
Send chunk 1:1 from message 1
Received chunk 1:1 from message 1
Received Message (id=1, head=[1,2,200], body={"result":["_system","testdb1","testdb2","repond"],"error":false,"code":200})
Send Message (id=2, head=[1,1,"testdb2",1,"\/_api\/collection",{},{}], body={})
Send chunk 1:1 from message 2
Received chunk 1:1 from message 2
Received Message (id=2, head=[1,2,200], body={"result":[{"id":"81395","name":"_apps","isSystem":true,"status":3,"type":2},{"id":"81387","name":"_aqlfunctions","isSystem":true,"status":3,"type":2},{"id":"81389","name":"_frontend","isSystem":true,"status":3,"type":2},{"id":"81370","name":"_graphs","isSystem":true,"status":3,"type":2},{"id":"81393","name":"_jobs","isSystem":true,"status":3,"type":2},{"id":"81372","name":"_modules","isSystem":true,"status":3,"type":2},{"id":"81391","name":"_queues","isSystem":true,"status":3,"type":2},{"id":"81374","name":"_routing","isSystem":true,"status":3,"type":2},{"id":"81421","name":"entities","isSystem":false,"status":3,"type":2},{"id":"81423","name":"docs2","isSystem":false,"status":3,"type":2},{"id":"81425","name":"edge1","isSystem":false,"status":3,"type":3},{"id":"81419","name":"types","isSystem":false,"status":3,"type":2}],"error":false,"code":200})
Collection types already exists
Send Message (id=3, head=[1,1,"testdb2",1,"\/_api\/collection",{},{}], body={})
Send chunk 1:1 from message 3
Received chunk 1:1 from message 3
Received Message (id=3, head=[1,2,200], body={"result":[{"id":"81395","name":"_apps","isSystem":true,"status":3,"type":2},{"id":"81387","name":"_aqlfunctions","isSystem":true,"status":3,"type":2},{"id":"81389","name":"_frontend","isSystem":true,"status":3,"type":2},{"id":"81370","name":"_graphs","isSystem":true,"status":3,"type":2},{"id":"81393","name":"_jobs","isSystem":true,"status":3,"type":2},{"id":"81372","name":"_modules","isSystem":true,"status":3,"type":2},{"id":"81391","name":"_queues","isSystem":true,"status":3,"type":2},{"id":"81374","name":"_routing","isSystem":true,"status":3,"type":2},{"id":"81421","name":"entities","isSystem":false,"status":3,"type":2},{"id":"81423","name":"docs2","isSystem":false,"status":3,"type":2},{"id":"81425","name":"edge1","isSystem":false,"status":3,"type":3},{"id":"81419","name":"types","isSystem":false,"status":3,"type":2}],"error":false,"code":200})
Collection entities already exists
Send Message (id=4, head=[1,1,"testdb2",1,"\/_api\/collection",{},{}], body={})
Send chunk 1:1 from message 4
Received chunk 1:1 from message 4
Received Message (id=4, head=[1,2,200], body={"result":[{"id":"81395","name":"_apps","isSystem":true,"status":3,"type":2},{"id":"81387","name":"_aqlfunctions","isSystem":true,"status":3,"type":2},{"id":"81389","name":"_frontend","isSystem":true,"status":3,"type":2},{"id":"81370","name":"_graphs","isSystem":true,"status":3,"type":2},{"id":"81393","name":"_jobs","isSystem":true,"status":3,"type":2},{"id":"81372","name":"_modules","isSystem":true,"status":3,"type":2},{"id":"81391","name":"_queues","isSystem":true,"status":3,"type":2},{"id":"81374","name":"_routing","isSystem":true,"status":3,"type":2},{"id":"81421","name":"entities","isSystem":false,"status":3,"type":2},{"id":"81423","name":"docs2","isSystem":false,"status":3,"type":2},{"id":"81425","name":"edge1","isSystem":false,"status":3,"type":3},{"id":"81419","name":"types","isSystem":false,"status":3,"type":2}],"error":false,"code":200})
Collection docs2 already exists
Send Message (id=5, head=[1,1,"testdb2",1,"\/_api\/collection",{},{}], body={})
Send chunk 1:1 from message 5
Received chunk 1:1 from message 5
Received Message (id=5, head=[1,2,200], body={"result":[{"id":"81395","name":"_apps","isSystem":true,"status":3,"type":2},{"id":"81387","name":"_aqlfunctions","isSystem":true,"status":3,"type":2},{"id":"81389","name":"_frontend","isSystem":true,"status":3,"type":2},{"id":"81370","name":"_graphs","isSystem":true,"status":3,"type":2},{"id":"81393","name":"_jobs","isSystem":true,"status":3,"type":2},{"id":"81372","name":"_modules","isSystem":true,"status":3,"type":2},{"id":"81391","name":"_queues","isSystem":true,"status":3,"type":2},{"id":"81374","name":"_routing","isSystem":true,"status":3,"type":2},{"id":"81421","name":"entities","isSystem":false,"status":3,"type":2},{"id":"81423","name":"docs2","isSystem":false,"status":3,"type":2},{"id":"81425","name":"edge1","isSystem":false,"status":3,"type":3},{"id":"81419","name":"types","isSystem":false,"status":3,"type":2}],"error":false,"code":200})
Collection edge1 already exists
Send Message (id=6, head=[1,1,"testdb2",1,"\/_api\/collection\/entities",{},{}], body={})
Send chunk 1:1 from message 6
Received chunk 1:1 from message 6
Received Message (id=6, head=[1,2,200], body={"id":"81421","name":"docs1","isSystem":false,"status":3,"type":2,"error":false,"code":200})
Send Message (id=7, head=[1,1,"testdb2",1,"\/_api\/index",{"collection":"docs1"},{}], body={})
Send chunk 1:1 from message 7
Received chunk 1:1 from message 7
Received Message (id=7, head=[1,2,200], body={"indexes":[{"id":"docs1\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},{"id":"docs1\/81429","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}],"identifiers":{"docs1\/0":{"id":"docs1\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},"docs1\/81429":{"id":"docs1\/81429","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}},"error":false,"code":200})
Send Message (id=8, head=[1,1,"testdb2",1,"\/_api\/index",{"collection":"docs1"},{}], body={})
Send chunk 1:1 from message 8
Received chunk 1:1 from message 8
Received Message (id=8, head=[1,2,200], body={"indexes":[{"id":"docs1\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},{"id":"docs1\/81429","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}],"identifiers":{"docs1\/0":{"id":"docs1\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},"docs1\/81429":{"id":"docs1\/81429","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}},"error":false,"code":200})
Collection docs1 already has a type index
Send Message (id=9, head=[1,1,"testdb2",1,"\/_api\/collection\/docs2",{},{}], body={})
Send chunk 1:1 from message 9
Received chunk 1:1 from message 9
Received Message (id=9, head=[1,2,200], body={"id":"81423","name":"docs2","isSystem":false,"status":3,"type":2,"error":false,"code":200})
Send Message (id=10, head=[1,1,"testdb2",1,"\/_api\/index",{"collection":"docs2"},{}], body={})
Send chunk 1:1 from message 10
Received chunk 1:1 from message 10
Received Message (id=10, head=[1,2,200], body={"indexes":[{"id":"docs2\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},{"id":"docs2\/81433","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}],"identifiers":{"docs2\/0":{"id":"docs2\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},"docs2\/81433":{"id":"docs2\/81433","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}},"error":false,"code":200})
Send Message (id=11, head=[1,1,"testdb2",1,"\/_api\/index",{"collection":"docs2"},{}], body={})
Send chunk 1:1 from message 11
Received chunk 1:1 from message 11
Received Message (id=11, head=[1,2,200], body={"indexes":[{"id":"docs2\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},{"id":"docs2\/81433","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}],"identifiers":{"docs2\/0":{"id":"docs2\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},"docs2\/81433":{"id":"docs2\/81433","type":"hash","fields":["type[*]"],"selectivityEstimate":1,"unique":false,"sparse":false}},"error":false,"code":200})
Collection docs2 already has a type index
Send Message (id=12, head=[1,1,"testdb2",1,"\/_api\/collection\/edge1",{},{}], body={})
Send chunk 1:1 from message 12
Received chunk 1:1 from message 12
Received Message (id=12, head=[1,2,200], body={"id":"81425","name":"edge1","isSystem":false,"status":3,"type":3,"error":false,"code":200})
Send Message (id=13, head=[1,1,"testdb2",1,"\/_api\/index",{"collection":"edge1"},{}], body={})
Send chunk 1:1 from message 13
Received chunk 1:1 from message 13
Received Message (id=13, head=[1,2,200], body={"indexes":[{"id":"edge1\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},{"id":"edge1\/1","type":"edge","fields":["_from","_to"],"selectivityEstimate":1,"unique":false,"sparse":false}],"identifiers":{"edge1\/0":{"id":"edge1\/0","type":"primary","fields":["_key"],"selectivityEstimate":1,"unique":true,"sparse":false},"edge1\/1":{"id":"edge1\/1","type":"edge","fields":["_from","_to"],"selectivityEstimate":1,"unique":false,"sparse":false}},"error":false,"code":200})
```
## Exception

```
com.arangodb.ArangoDBException: com.arangodb.velocypack.exception.VPackParserException: java.lang.IllegalArgumentException: No enum constant com.arangodb.entity.IndexType.edge
	at com.arangodb.util.ArangoUtil.deserialize(ArangoUtil.java:72)
	at com.arangodb.internal.ArangoExecutor.deserialize(ArangoExecutor.java:112)
	at com.arangodb.internal.InternalArangoCollection$9.deserialize(InternalArangoCollection.java:512)
	at com.arangodb.internal.InternalArangoCollection$9.deserialize(InternalArangoCollection.java:509)
	at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:59)
	at com.arangodb.ArangoCollection.getIndexes(ArangoCollection.java:625)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.MethodMetaProperty$GetBeanMethodMetaProperty.getProperty(MethodMetaProperty.java:76)
	at org.codehaus.groovy.runtime.callsite.GetEffectivePojoPropertySite.getProperty(GetEffectivePojoPropertySite.java:64)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:296)
	at io.repond.entities.ArangoEntityStore$_initTypeIndexes_closure1.doCall(ArangoEntityStore.groovy:189)
  ....
Caused by: com.arangodb.velocypack.exception.VPackParserException: java.lang.IllegalArgumentException: No enum constant com.arangodb.entity.IndexType.edge
	at com.arangodb.velocypack.VPack.deserialize(VPack.java:299)
	at com.arangodb.util.ArangoUtil.deserialize(ArangoUtil.java:68)
	... 79 more
Caused by: java.lang.IllegalArgumentException: No enum constant com.arangodb.entity.IndexType.edge
	at java.lang.Enum.valueOf(Enum.java:238)
	at com.arangodb.velocypack.VPack.getValue(VPack.java:427)
	at com.arangodb.velocypack.VPack.deserializeField(VPack.java:391)
	at com.arangodb.velocypack.VPack.deserializeFields(VPack.java:379)
	at com.arangodb.velocypack.VPack.deserializeObject(VPack.java:330)
	at com.arangodb.velocypack.VPack.getValue(VPack.java:429)
	at com.arangodb.velocypack.VPack.deserializeCollection(VPack.java:457)
	at com.arangodb.velocypack.VPack.getValue(VPack.java:413)
	at com.arangodb.velocypack.VPack.deserialize(VPack.java:297)
	... 80 more
```

**Comment by mpv1989** [comment]
Hi,
this is a bug in the java driver. I will release a new version (4.1.9) within this week. In the meantime you can use the snapshot `4.1.9-SNAPSHOT` which already inlcudes the fix.

Best
Mark

**Comment by dan-drew** [comment]
Thanks @mpv1989 I'll try that out

**Comment by mpv1989** [comment]
released version `4.1.9` which includes the fix
